flow360/component/simulation/outputs/output_entities.py

```diff
@@ -112,6 +113,22 @@ def _preprocess_expression_and_solver_variable(cls, value):
             )
         return solver_variable_to_user_variable(value)
 
+    @pd.field_validator("iso_value", mode="before")
+    @classmethod
+    def _preprocess_field_with_unit_system(cls, value, info: pd.ValidationInfo):
+        if not isinstance(value, dict) or info.data.get("field") is None or "units" not in value:
+            return value
+        units = value["units"]
+        field = info.data["field"]
+        field_dimensions = get_input_value_dimensions(value=field)
+        if units in ("SI", "SI_unit_system"):
```

Comment:
```suggestion
        if units in  "SI_unit_system":
```
This is a UI only feature and we should limit the allowed input.